File: Elixir/ecto/lib/models/department.ex
defmodule Models.Department do
  use Ecto.Schema

  schema "departments" do
    field(:name, :string)
  end

  def changeset(department, params \\ %{}) do
    department
    |> Ecto.Changeset.cast(params, [:name])
    |> Ecto.Changeset.validate_required([:name])
  end
end

defimpl String.Chars, for: Models.Department do
  def to_string(dep) do
    "Department { Id: #{dep.id}, Name: #{dep.name} }"
  end
end

File: FSharp/EntityFramework/Models/Department.fs
namespace Models

open System
open System.ComponentModel.DataAnnotations

type Department () =
  [<Key>]
  member val Id: int = Int32 () with get, set
  member val Name: string = null with get, set

  new (name) as this =
    Department ()
    then
      this.Name <- name

  with override this.ToString () =
        $"Department {{ Id: {this.Id}, Name: {this.Name} }}"

